bio_hansel/aho_corasick/__init__.py

```diff
@@ -1,12 +1,60 @@
 # -*- coding: utf-8 -*-
 
 from collections import defaultdict
+from itertools import product
+import logging
 
 from ahocorasick import Automaton
 import pandas as pd
 
 from ..parsers import parse_fasta, parse_fastq
 from ..utils import revcomp
+from ..const import bases_dict
+from ..subtyping_params import SubtypingParams
+
+def expand_degenerate_bases(seq):
+    """List all possible kmers for a scheme given a degenerate base
+   
+    Args:
+         Scheme_kmers from SNV scheme fasta file
+
+    Returns:
+         List of all possible kmers given a degenerate base or not
+    """
+
+    return list(map("".join, product(*map(bases_dict.get, seq))))
+
+def check_total_kmers(scheme_fasta, subtyping_params):
+    '''Checks that the number of kmers about to be created is not at too high a computation or time cost
+
+    Args:
+         scheme_fasta: Kmer sequences from the SNV scheme
+         Subtyping parameter max_degenerate_kmers: The max kmers allowed by the scheme
+    
+    Returns:
+         None if created kmers < max degenerate kmers argument
+         Exits code with warning if created kmers > max degenerate kmers argument
+    '''
+    kmer_number = 0
+    for header, sequence in parse_fasta(scheme_fasta):
+        value = 1
+        for char in sequence:
+            length_key = len(bases_dict[char])
+            value = value * length_key
+        kmer_number = kmer_number + value
+    if kmer_number*2 > subtyping_params.max_degenerate_kmers:
+        return logging.error(
+            '''
+    Your current scheme contains "{}" kmers which is over the reccomended number of "{}".
```

Comment:
Maybe change wording that the 'reccomended number' is either the default value OR what was passed to max-degenerate-kmers' flag. 

Also typo  'reccomended'